### PR TITLE
Merge onnx_importer to onnx_ngraph_frontend

### PR DESCRIPTION
**Pull request by mateusztabaka**
Ticket: 60948

### CODE REVIEW

File: cmake/features.cmake [hunk 1]
```diff
@@ -122,8 +122,7 @@ else()
     set(protoc_available ON)
 endif()
 
-ie_dependent_option(NGRAPH_ONNX_IMPORT_ENABLE "Enable ONNX importer" ON "protoc_available" OFF)
-ie_dependent_option(NGRAPH_ONNX_FRONTEND_ENABLE "Enable ONNX FrontEnd" OFF "NGRAPH_ONNX_IMPORT_ENABLE" OFF)
+ie_dependent_option(NGRAPH_ONNX_IMPORT_ENABLE "Enable ONNX FrontEnd" ON "protoc_available" OFF)
```
Comment (ilya-lavrenov) [inline cmake/features.cmake:125]: may be better to keep  `NGRAPH_ONNX_FRONTEND_ENABLE` name since importer is obsolete name? currently we operate with frontends

File: docs/IE_DG/Extensibility_DG/Custom_ONNX_Ops.md [hunk 2]
```diff
@@ -50,9 +50,9 @@ The example below demonstrates how to unregister an operator from the destructor
 
 ## Requirements for Building with CMake
 
-A program that uses the `register_operator` functionality requires `ngraph` and `onnx_importer` libraries in addition to the Inference Engine.
-The `onnx_importer` is a component of the `ngraph` package , so `find_package(ngraph REQUIRED COMPONENTS onnx_importer)` can find both.
-The `ngraph` package exposes two variables, `${NGRAPH_LIBRARIES}` and `${ONNX_IMPORTER_LIBRARIES}`, which reference the `ngraph` and `onnx_importer` libraries.
+A program that uses the `register_operator` functionality requires `ngraph` and `onnx_ngraph_frontend` libraries in addition to the Inference Engine.
+The `onnx_ngraph_frontend` is a component of the `ngraph` package , so `find_package(ngraph REQUIRED COMPONENTS onnx_ngraph_frontend)` can find both.
+The `ngraph` package exposes two variables, `${NGRAPH_LIBRARIES}` and `${ONNX_FRONTEND_LIBRARIES}`, which reference the `ngraph` and `onnx_ngraph_frontend` libraries.
```
Comment (ilya-lavrenov) [inline docs/IE_DG/Extensibility_DG/Custom_ONNX_Ops.md:55]: can we remove `ONNX_FRONTEND_LIBRARIES` variables and work with importer targets directly?

Also, I believe we need to add suffix `ngraph::` to all importer targets

File: ngraph/frontend/onnx/frontend/include/onnx_import/core/null_node.hpp [hunk 3]
```diff
@@ -31,7 +31,7 @@ namespace ngraph
         ///
         /// More:
         /// https://github.com/onnx/onnx/blob/master/docs/IR.md#optional-inputs-and-outputs
-        class NullNode : public ngraph::Node
+        class ONNX_IMPORTER_API NullNode : public ngraph::Node
```
Comment (ilya-lavrenov) [inline ngraph/frontend/onnx/frontend/include/onnx_import/core/null_node.hpp:34]: why is it migrated from private API?

File: docs/onnx_custom_op/CMakeLists.txt [hunk 4]
```diff
@@ -7,11 +7,12 @@ set(CMAKE_CXX_STANDARD 11)
 
 set(TARGET_NAME "onnx_custom_op")
 
-find_package(ngraph REQUIRED COMPONENTS onnx_importer)
+find_package(ngraph REQUIRED COMPONENTS onnx_ngraph_frontend)
```
Comment (ilyachur) [inline docs/onnx_custom_op/CMakeLists.txt:10]: Maybe can we simplify the name? `onnx_frontend`?

Comment (ilyachur) [reply docs/onnx_custom_op/CMakeLists.txt:10]: @nosovmik The same question for PDPD frontend.

File: ngraph/frontend/onnx/frontend/include/onnx_import/core/null_node.hpp (hunk 3 above)
Comment (ilyachur) [reply ngraph/frontend/onnx/frontend/include/onnx_import/core/null_node.hpp:34]: I guess it should be a part of private API

File: docs/onnx_custom_op/CMakeLists.txt (hunk 4 above)
Comment (nosovmik) [reply docs/onnx_custom_op/CMakeLists.txt:10]: Frontends are discovered by FrontEndManager using OV_FRONTEND_PATH environment variable. FE Manager searches for "*_ngraph_frontend" libraries on this location and tries to load them. Mask "*_frontend" looked too generic for such search

Comment (ilya-lavrenov) [reply docs/onnx_custom_op/CMakeLists.txt:10]: _ov_frontend?

Comment (nosovmik) [reply docs/onnx_custom_op/CMakeLists.txt:10]: Yes, `*_ov_frontend` looks also fine. If you think this is more suitable, let's create ticket and correct this. It is better to do under separate PR as it will require updates of 'mock test' frontend names as well

File: ngraph/frontend/onnx/frontend/include/onnx_import/core/null_node.hpp (hunk 3 above)
Comment (mateusztabaka) [reply ngraph/frontend/onnx/frontend/include/onnx_import/core/null_node.hpp:34]: I think it should be public. It was overlooked before, but we need that for custom onnx ops mechanism. NullNode is the way to represent missing optional input or output and I think we should allow users to use it like we do e.g. in batch norm https://github.com/openvinotoolkit/openvino/blob/master/ngraph/frontend/onnx/onnx_import/src/op/batch_norm.cpp#L38

File: ngraph/test/CMakeLists.txt [hunk 5]
```diff
@@ -635,11 +635,14 @@ if (OV_COMPILER_IS_CLANG)
...
-    target_include_directories(unit-test PRIVATE ${ONNX_IMPORTER_SRC_DIR}/src)
+if (NGRAPH_ONNX_FRONTEND_ENABLE)
+    get_target_property(ONNX_FRONTEND_SRC_DIR onnx_ngraph_frontend SOURCE_DIR)
+    target_include_directories(unit-test PRIVATE ${ONNX_FRONTEND_SRC_DIR}/src)
 
-    target_link_libraries(unit-test PRIVATE onnx_importer)
+    target_link_libraries(unit-test PRIVATE onnx_ngraph_frontend)
+    if (LINUX)
+        target_link_options(unit-test PRIVATE -Wl,--exclude-libs,ALL)
+    endif()
```
Comment (ilya-lavrenov) [inline ngraph/test/CMakeLists.txt:645]: why do we need it? can we remove this?

Comment (mateusztabaka) [reply ngraph/test/CMakeLists.txt:645]: This is to not export symbols from onnx_common (in unit-test binary) so they're not bound do libonnx_ngraph_frontend.so during relocation. Since we switched protobuf from a dynamic to static library we need to hide its symbols in unit-test so libonnx_ngraph_frontend can use its own protobuf copy.

File: ngraph/frontend/onnx/frontend/include/onnx_import/core/null_node.hpp (hunk 3 above)
Comment (ilyachur) [reply ngraph/frontend/onnx/frontend/include/onnx_import/core/null_node.hpp:34]: Before the NullNode was hidden and it was a private API.
Looks like in the case of optional inputs it can be needed for majority frameworks.
@slyalin Could you comment?

**Review by mbencer — APPROVED** [review]

File: .ci/azure/linux_onnxruntime.yml [hunk 6]
```diff
@@ -67,6 +67,7 @@ jobs:
       python3 -m pip install -r $(REPO_DIR)/inference-engine/ie_bridges/python/requirements.txt
       # For running Python API tests
       python3 -m pip install -r $(REPO_DIR)/inference-engine/ie_bridges/python/src/requirements-dev.txt
+      python3 -m pip install cmake>=3.18
```
Comment (ilya-lavrenov) [inline .ci/azure/linux_onnxruntime.yml:70]: why do we need non-default cmake?

Comment (mateusztabaka) [reply .ci/azure/linux_onnxruntime.yml]: that was required for onnxruntime build. But it's not anymore since the onnxruntime branch was fixed to 1.8.1

File: ngraph/frontend/onnx/frontend/include/onnx_import/core/null_node.hpp (hunk 3 above)
Comment (ilyachur) [reply ngraph/frontend/onnx/frontend/include/onnx_import/core/null_node.hpp:34]: @slyalin ping

I just want to know can this operation be used by other frontends? Should we move it to common part?

File: .ci/azure/linux_onnxruntime.yml [hunk 7]
```diff
@@ -114,6 +113,7 @@ jobs:
 
   - script: |
       source $(INSTALL_DIR)/bin/setupvars.sh
+      echo "2022.1" > $(INSTALL_DIR)/deployment_tools/inference_engine/version.txt
```
Comment (ilyachur) [inline .ci/azure/linux_onnxruntime.yml:116]: I don't think that we need this change.

File: docs/onnx_custom_op/CMakeLists.txt (hunk 4 above)
Comment (ilyachur) [reply docs/onnx_custom_op/CMakeLists.txt:10]: Can we rename to `ov_frontend`?

Comment (nosovmik) [reply docs/onnx_custom_op/CMakeLists.txt:10]: It will need renaming of other frontends (like paddlepaddle and several frontends for testing purposes). Created ticket 61780 for that

**Comment by azhogov** [comment]
CI passed: https://github.com/intel-innersource/frameworks.ai.openvino.ci.product-configs/pull/78

File: .ci/azure/linux_onnxruntime.yml [hunk 8]
```diff
@@ -57,7 +57,7 @@ jobs:
 
   - script: |
       branch=`tr -s '\n ' < $(ONNXRUNTIME_UTILS)/version`
-      git clone --branch $branch --single-branch --recursive https://github.com/microsoft/onnxruntime.git $(ONNXRUNTIME_REPO_DIR)
+      git clone --branch $branch --single-branch --recursive https://github.com/intel/onnxruntime.git $(ONNXRUNTIME_REPO_DIR)
     displayName: 'Clone onnxruntime'
```
Comment (ilya-lavrenov) [inline .ci/azure/linux_onnxruntime.yml:61]: I prefer to contribute to ONNX runtime instead of having our fork

**Comment by ilyachur** [comment]
I don't want to merge this PR while we are using ONNXRuntime fork instead of original repo

File: .ci/azure/linux_onnxruntime.yml (hunk 8 above)
Comment (ilya-lavrenov) [reply .ci/azure/linux_onnxruntime.yml:61]: BTW, the only change in https://github.com/intel/onnxruntime/commit/ea0b3e693ded146099652fc281c58c1a5e39c004 I see it `onnx_ngraph_frontend` instead of `onnx_importer`. Can we create an alias target? In this case we will have both names for BW compatibility

File: ngraph/frontend/onnx/frontend/CMakeLists.txt [hunk 9]
```diff
@@ -25,19 +37,24 @@ if(COMMAND ie_add_vs_version_file)
...
 
 target_include_directories(onnx_ngraph_frontend PUBLIC $<BUILD_INTERFACE:${ONNX_FRONTEND_INCLUDE_DIR}>
                                                 $<INSTALL_INTERFACE:${FRONTEND_INSTALL_INCLUDE}>)
-
 target_include_directories(onnx_ngraph_frontend PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/src)
 
+target_compile_definitions(onnx_ngraph_frontend PRIVATE ONNX_OPSET_VERSION=${ONNX_OPSET_VERSION})
+if(NGRAPH_USE_PROTOBUF_LITE)
+    target_compile_definitions(onnx_ngraph_frontend PRIVATE NGRAPH_USE_PROTOBUF_LITE)
+endif()
```
Comment (ilyachur) [inline ngraph/frontend/onnx/frontend/CMakeLists.txt:49]: Can we introduce alias to onnx_importer for backward compatibility?

Comment (mbencer) [reply ngraph/frontend/onnx/frontend/CMakeLists.txt:49]: As I know @mateusztabaka (he is OOO today) try to use an alias, but with failure.
In cmake spec we can find:
```ALIAS targets are not mutable, installable or exportable. They are entirely local to the buildsystem description```

Comment (ilya-lavrenov) [reply ngraph/frontend/onnx/frontend/CMakeLists.txt:49]: We need to create alias in `ngraphConfig.cmake`, not in source tree :)

**Review by ilyachur — APPROVED** [review]

File: ngraph/frontend/onnx/frontend/CMakeLists.txt [hunk 10]
```diff
@@ -20,29 +32,35 @@ add_library(ngraph::onnx_ngraph_frontend ALIAS onnx_ngraph_frontend)
 
 add_clang_format_target(onnx_ngraph_frontend_clang FOR_TARGETS onnx_ngraph_frontend)
 
-ov_ncc_naming_style(FOR_TARGET onnx_ngraph_frontend
-                    INCLUDE_DIRECTORY "${ONNX_FRONTEND_INCLUDE_DIR}"
-                    ADDITIONAL_INCLUDE_DIRECTORIES
-                        $<TARGET_PROPERTY:ngraph::frontend_manager::static,INTERFACE_INCLUDE_DIRECTORIES>)
+# TODO: fix empty class name
```
Comment (ilyachur) [inline ngraph/frontend/onnx/frontend/CMakeLists.txt:35]: Please fix naming style separately

Comment (ilya-lavrenov) [reply ngraph/frontend/onnx/frontend/CMakeLists.txt:35]: See https://github.com/openvinotoolkit/openvino/pull/7033
